ironfish/src/assets/assetsVerifier.ts

```diff
@@ -80,6 +86,7 @@ export class AssetsVerifier {
       }
     } catch (error) {
       this.logger.warn(`Error while fetching verified assets: ${ErrorUtils.renderError(error)}`)
+      throw error
```

Comment:
What type of error could happen here, besides network connection issue?  It will be good if we only retry when the error is from network issue or other retry-able error.

Reply:
This could be a network error, or a server error (e.g. a 500 or 521 response), or an invalid/truncated JSON payload. The HTTP method used here (GET) is both [safe](https://www.rfc-editor.org/rfc/rfc7231#section-4.2.1) and [idempotent](https://www.rfc-editor.org/rfc/rfc7231#section-4.2.2), so this kind of request is always retry-able, regardless of the kind of error returned.

Reply:
For the internal server error 500, would it still be failing when calling `refreshVerifiedAssets` when retry again?

Reply:
The next call to `refreshVerifiedAssets` may or may not fail: it all depends on whether the remote API has recovered or not.